src/util/deferred.ts

```diff
@@ -0,0 +1,29 @@
+export class Deferred<T> {
+  private readonly _promise: Promise<T>
```

Comment:
You can (and should) use real private properties. The minimum supported node version already supports private properties and they are used extensively in Kysely.